In [1]:
import numpy as np
import pandas as pd

In [2]:
# The statistical module used to run chi square test
import scipy.stats as stats

In [4]:
# Read csv file
cycle_df = pd.read_csv("output_data/cyclingstreets_central_london.csv")

# Display data
cycle_df.head()

Unnamed: 0     type             ID                  Date&Time Severity  \
0           0  Feature  199901AB00008   5:15pm, 7th January 1999  serious   
1           1  Feature  199901AB00018  9:20am, 11th January 1999   slight   
2           2  Feature  199901AB00019  3:00pm, 12th January 1999   slight   
3           3  Feature  199901AB00021  11:00am, 9th January 1999   slight   
4           4  Feature  199901AB00023  8:12pm, 19th January 1999   slight   

  Casualties  # of Casualties  # of Vehicles   Latitude  Longitude  \
0    Cyclist                1              1  51.496219  -0.134989   
1    Cyclist                1              2  51.489624  -0.127480   
2    Cyclist                1              2  51.489951  -0.148065   
3    Cyclist                1              2  51.490698  -0.149764   
4    Cyclist                1              2  51.494715  -0.125254   

              Coordinates  
0  [-0.134989, 51.496208]  
1  [-0.127474, 51.489641]  
2   [-0.14809, 51.489926]  
3    [-0.14978, 51.49069]  
4  [-0.125271, 51.494692]

In [7]:
a = len(cycle_df.loc[cycle_df['Severity'] == "slight"])
b = len(cycle_df.loc[cycle_df['Severity'] == "serious"])
c = len(cycle_df.loc[cycle_df['Severity'] == "fatal"])
print(a,b,c)
len(cycle_df)

24442 3957 107


28506

In [8]:
# Observed data in a (hypothetical) survey of 300 people 
observed = pd.Series([a,b,c], index=["slight", "serious", "fatal"])

In [9]:
# Create a data frame
df = pd.DataFrame([observed]).T

In [10]:
# Add a column whose default values are the expected values
df[1] = 9502

In [11]:
# Rename columns
df.columns = ["observed", "expected"]

In [12]:
# View the data frame
df

observed  expected
slight      24442      9502
serious      3957      9502
fatal         107      9502

In [13]:
# The degree of freedom is 3-1 = 2
# With a p-value of 0.05, the confidence level is 1.00-0.05 = 0.95.
critical_value = stats.chi2.ppf(q = 0.95, df = 2)

In [14]:
# The critical value
critical_value

5.991464547107979

In [15]:
# Run the chi square test with stats.chisquare()
stats.chisquare(df['observed'], df['expected'])

Power_divergenceResult(statistic=36015.22311092402, pvalue=0.0)

## Conclusion

Since the chi square value of 36015.22 exceeds the critical value of 5.99, we conclude that the results are statistically significant.